### PR TITLE
ComboBox virtualization, item kidnapping fixes

### LINKED ISSUE #555: wasm: ComboBox is unreasonably slow
## I'm submitting a...

 - Feature request 

## Current behavior
Using the repo at 
A UWP combobox with 2000 items can have its datacontext set in 0.1s
The same code in wasm takes 90 seconds (debugger not open).
In the DOM it hasn't created the popup items yet which is good, so what its doing is unknown, and seems unnecessary.  In a LOB app combos are often used to lookup between keys and names and sometimes never opened so doing preparation (if that's what it's doing) for opening is not wanted and inefficient.

## Expected behavior
When not opening the combo, performance should be acceptable even in interpreted mode, after all the screen just has an empty combo

## Minimal reproduction of the problem with instructions
download https://github.com/yowl/UnoStandardRefTest/tree/comboperf and run in UWP and wasm and observe the performance.

## Environment

```
Nuget Package: 
    <PackageReference Include="Uno.UI" Version="1.43.0-dev.746" />
    <PackageReference Include="Uno.Wasm.Bootstrap" Version="1.0.0-dev.159" />
Package Version(s): 

Affected platform(s):
- [ ] iOS
- [ ] Android
- [ x] WebAssembly
- [x ] Windows
- [ ] Build tasks

Visual Studio
- [ x] 2017 (version: )
- [ ] 2017 Preview (version: )
- [ ] for Mac (version: )

Relevant plugins
- [ ] Resharper (version: )
```

### LINKED ISSUE #1078: ComboBox support for view items is glitchy
## How to reproduce it (as minimally and precisely as possible)

1. https://github.com/nventive/Uno/blob/master/src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/ComboBox/ComboBox_ComboBoxItem_Selection.xaml

2. Select any item in the second dropdown.

3. Open the dropdown again.

4. Dismiss the dropdown without modifying the selection.

## Current behavior

Selected item has disappeared.

## Expected behavior

Display of selected item is robustly maintained even when items are views.

## Environment

Nuget Package: Uno.UI

Package Version(s): 1.44.1

Affected platform(s):
- [x] iOS
- [x] Android
- [x] WebAssembly
- [ ] Windows
- [ ] Build tasks

Visual Studio
- [ ] 2017 (version: )
- [ ] 2019 (version: )
- [ ] for Mac (version: )

Relevant plugins
- [ ] Resharper (version: )

## Anything else we need to know?

Related to #87

### PR DESCRIPTION
**Pull request by davidjohnoliver**
GitHub Issue (If applicable): Fixes #555, partially addresses #556 and #1133, fixes #1078

## PR Type
What kind of change does this PR introduce?

 - Bugfix 
 - Feature 

## What is the new behavior?

 - [WASM] ComboBox is now virtualized
 - [WASM] ListView supports non-Stretch alignments properly.
 - [All] Fixes issues with views in `Items`/`ItemsSource` being 'kidnapped' when closing and opening `ComboBox` dropdown.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by carldebilly** [comment]
19 commits for that ? ;-)

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/ListView/ListView_Aligned_Left.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+using Windows.Foundation.Collections;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Navigation;
+
+// The User Control item template is documented at https://go.microsoft.com/fwlink/?LinkId=234236
```
Comment (carldebilly) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/ListView/ListView_Aligned_Left.xaml.cs]: Usually I like to clean that up...

**Comment by davidjohnoliver** [comment]
> 19 commits for that ? ;-)

You've already reviewed most of them. ;) It includes #1061 which had to be reverted, and also #1079

**Comment by davidjohnoliver** [comment]
TODO: Add UI tests for 'kidnapping'

**Comment by carldebilly** [comment]
`Error CS1061: 'FrameworkElement' does not contain a definition for 'GetVisualTreeParent'`

**Pull request merged**